### DISCUSSION TITLE
signoz helm install fails, need help in debugging the issue

### DISCUSSION THREAD

**Discussion by Yeshwanth-git2022** [SigNoz/signoz#5462] [Q&A]
![image](https://github.com/SigNoz/signoz/assets/98457676/ecdc4534-118a-409c-bc93-ecd082bdaa28)
which service should I start debugging first?
I verified the pre-requisite and other installs like docker and helm that are required.

**Comment by Yeshwanth-git2022** [comment]
I started my analysis with the chi-my-release-clickhouse-cluster-0-0-0 pod, since query service depends on it. Following is what I see in the logs for the init container that is having the issue. 
![image](https://github.com/SigNoz/signoz/assets/98457676/9735e513-2b73-4d69-a628-fc5aae5a0b2d)
I have proxy configured on my vm and i can run the following command without any issues.
wget -O /tmp/histogramQuantile https://github.com/SigNoz/signoz/raw/develop/deploy/docker/clickhouse-setup/user_scripts/histogramQuantile
But the docker running the clickhouse seems to have an issue.

**Reply by arathore47** [reply]
@Yeshwanth-git2022 Is this the full screenshot? I only see two lines and not any error

**Reply by Yeshwanth-git2022** [reply]
thanks for the reply. container is stuck on the line - connecting to GitHub. After the timeout, process exits and no progress after that.

**Reply by arathore47** [reply]
Hm, Can you add "sleep infinity" as the cmd in the chi-my-release-clickhouse-cluster-0-0-0 pod and then try wget after going inside the pod through exec? If that fails, can you try netcat on the same IP from inside? Those results might help us

**Reply by Zarloc** [reply]
I have the same issue, it's linked to an unresolved issue between alpine and github.com DNS records cf: https://github.com/alpinelinux/docker-alpine/issues/155
![image](https://github.com/user-attachments/assets/263de08c-60b3-4413-89d9-a5f054ce4365)

Workaround use busybox image based on libc and not musl
```yaml
clickhouse:
  initContainers:
    udf:
      image:
        repository: busybox
        tag: latest
```

**Reply by arathore47** [reply]
@Zarloc If this is not production, Can you try to restart the coredns/kubedns pod on kube-system? That solved the issue for me earlier, let me know if it works.

**Reply by Zarloc** [reply]
Actually it's a minikube on Docker (Ubuntu 24.04), same error after a restart of coredns or a fresh minikube instance.

I don't yet, tested the chart on my EKS cluster.

**Reply by Zarloc** [reply]
Here Minikube + Docker + NetworkManager (Ubuntu) + coworking building network are a pretty exotic/non-prod environment, but this same issue (alpine) already occur on our old RKE cluster directly plug to internet via BGP router.

**Reply by Yeshwanth-git2022** [reply]
I tried both restarting coredns as well as updating the image of udf init container to busybox:latest. same error. still stuck at the same place.

**Reply by arathore47** [reply]
- Is the github url resolvable through nslookup/curl when you exec into the clickhouse pod? I would be surprised if it is.
- If no, is the same URL resolvable inside any other pod in the cluster? If yes, this is a general cluster wide DNS service issue

@Yeshwanth-git2022 @Zarloc

**Reply by Yeshwanth-git2022** [reply]
Tried adding sleep infinity and running wget inside the container. still the same issue. I will try to investigate further on the proxy.

![image](https://github.com/user-attachments/assets/9e571c68-7abb-4d8b-9bfe-85f3f1687ade)

**Reply by Yeshwanth-git2022** [reply]
github.com is resolvable
![image](https://github.com/user-attachments/assets/512ce9a5-94c0-4cc0-ab67-7bff54e529f5)

**Reply by arathore47** [reply]
@Yeshwanth-git2022 from inside clickhouse, try to download any sample file through the internet through wget, if it succeeds, try the same wget command for the github file on your local, the combined results should give us further clues

**Reply by Yeshwanth-git2022** [reply]
ok I was able to get it to work by setting the proxy in the env section and using apline/curl:latest image and replacing wget -O with curl -o option.
